### PR TITLE
jsonresp: remove top level error key, allow caller to define the entire error json response

### LINKED ISSUE #6: allow caller to define the entire error json response
> but if app codebase is returning a thing to be json encoded, it's better to let them define the entire json and not unnecessarily insist on a single top level `"error"` json key

_Originally posted by @choonkeat in https://github.com/alvinchoong/go-httphandler/pull/5#discussion_r2090725394_

### PR DESCRIPTION
**Pull request by alvinchoong**
## Summary

This PR removes the constraint that error responses must have a top-level "error" key. Instead, it allows users to define their entire JSON structure for error responses (just like success responses), providing more flexibility.

## Changes

- Refactored `errorResponder` to use the provided data directly without wrapping it in a top-level "error" key
- Enhanced test coverage with more comprehensive test cases for various data types
- Updated examples and documentation to reflect the new approach

## Motivation

When integrating with existing codebases, forcing a specific top-level "error" key can be restrictive. This change allows users to Define their own error response structure completely

## Breaking Changes

Before:

```go
// Using a structured error response
type ValidationError struct {
    Code    string `json:"code"`
    Message string `json:"message"`
    Details string `json:"details,omitempty"`
}

// Error data is automatically wrapped with a top-level "error" key
jsonresp.Error(err, ValidationError{
    Code:    "VALIDATION_ERROR",
    Message: "Invalid input",
    Details: err.Error(),
}, http.StatusBadRequest)

// Response: {"error": {"code": "VALIDATION_ERROR", "message": "Invalid input", "details": "..."}}
```

After:

```go
// Using the same structured error response
type ValidationError struct {
    Code    string `json:"code"`
    Message string `json:"message"`
    Details string `json:"details,omitempty"`
}

// Error data is used directly without wrapping
jsonresp.Error(err, &ValidationError{
    Code:    "VALIDATION_ERROR",
    Message: "Invalid input",
    Details: err.Error(),
}, http.StatusBadRequest)

// Response: {"code": "VALIDATION_ERROR", "message": "Invalid input", "details": "..."}
```

## Testing

Added comprehensive test cases covering various data types and edge cases.

## Migration Guide

To migrate from the previous version:

Simply wrap the error message in a struct or map:

```go
// Before
jsonresp.Error(err, "Error message", statusCode)

// After - Option 1: Using a map
jsonresp.Error(err, &map[string]string{
    "error": "Error message"
}, statusCode)

// After - Option 2: Using a struct
type ErrorResponse struct {
    Error string `json:"error"`
}

jsonresp.Error(err, &ErrorResponse{Error: "Error message"}, statusCode)
```

## Related Issues

Closes #6

### CODE REVIEW

File: jsonresp/error.go [hunk 1]
```diff
@@ -54,7 +55,7 @@ func (res *errorResponder[T]) Respond(w http.ResponseWriter, _ *http.Request) {
 	}
 
 	// Write the error JSON response.
-	writeJSON(w, map[string]T{"error": res.errData}, res.statusCode, res.logger)
```
Comment (alvinchoong) [inline jsonresp/error.go:57]: main change, remove top level key

File: jsonresp/error_test.go [hunk 2]
```diff
@@ -28,33 +40,72 @@ func TestError_Respond(t *testing.T) {
...
+			wantCookies: nil,
+			wantBody:    `"Invalid ID provided"`,
+		},
+		{
+			desc:        "basic | slice",
+			given:       jsonresp.Error(errors.New("invalid id"), &[]string{"Required field", "Invalid format"}, http.StatusBadRequest),
 			wantCode:    http.StatusBadRequest,
 			wantHeaders: nil,
 			wantCookies: nil,
-			wantBody:    `{"error":"Invalid ID provided"}`,
```
Comment (choonkeat) [inline jsonresp/error_test.go:37]: so this isn't [a backwards compatible change](https://github.com/alvinchoong/go-httphandler/pull/5#pullrequestreview-2842961647) 👌

**Review by choonkeat — APPROVED** [review]

File: jsonresp/error_test.go (hunk 2 above)
Comment (alvinchoong) [reply jsonresp/error_test.go:37]: Yep, unfortunately
